services/server: use EOF exception

viktorTarasov opened the pull request:
In KMIPProtocol use 'EOFError' exception when there are no more bytes to read from input stream.
This allows to distinguish the 'end of transaction' from 'unexpected length of data read' and to avoid the error log at the end of transaction.

PeterHamilton commented:
I like improving the error handling for the socket data. See my comments above and this should be good to go.

viktorTarasov commented:
- implemented the _RequestLengthMismatch_ exception as you suggested.
- the _EOFException_ is raised only when reading _Request.HEADER_.
- added unit tests for kmip-protocol to pass the _travis coverage/coveralls_ tests

viktorTarasov commented:
tests are re-implemented with _mock_

joel-coffman commented on kmip/services/server/kmip_protocol.py:
```diff
@@ -63,3 +72,22 @@ class KMIPProtocolFactory(object):
+
+class RequestLengthMismatch(Exception):
+    """
+    An exception raised for error when the request read from stream
```
nit: "for error" is unnecessary in my opinion

joel-coffman commented on kmip/services/server/kmip_server.py:
```diff
@@ -53,6 +53,7 @@ def serve(self):
         self.socket.listen(0)
         while True:
             connection, address = self.socket.accept()
+            self.logger.info('New connection')
```
If adding a logging statement, is it possible to include more detailed information about the connection that would be useful when debugging?

joel-coffman commented on kmip/services/server/kmip_server.py:
```diff
@@ -70,9 +71,13 @@ def serve(self):
                 self.logger.error('KMIPServer {0} {1}'.format(type(e), e))
-                connection.close()
+
+            connection.close()
```
Matter of style I suppose, but I prefer for `connection.close()` to be part of a `finally` block.

viktorTarasov commented:
* fixed spelling of comments;
* _close.connection_ moved to _finaly_ block;
* _address_ printed in debug message for new connection

viktorTarasov commented:
little refactoring of tests

viktorTarasov commented:
rebased on actual _master_
